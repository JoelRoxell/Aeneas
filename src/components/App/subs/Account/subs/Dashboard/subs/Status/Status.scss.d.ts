// This file is generated automatically
export const center: string;
export const status: string;
export const circleWrapper: string;
export const circle: string;
export const iconWrapper: string;
export const icon: string;
export const ratio: string;
export const title: string;
