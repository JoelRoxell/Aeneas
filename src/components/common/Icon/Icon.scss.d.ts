// This file is generated automatically
export const center: string;
export const icon: string;
export const badge: string;
