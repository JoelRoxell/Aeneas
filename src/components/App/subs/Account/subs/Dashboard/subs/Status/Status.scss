.status {
  @extend .center;

  .circle-wrapper {
    $width: 76;

    position: relative;
    @extend .center;

    .circle {
      width: rem($width);
      height: rem($width);
      border-radius: rem($width/2);
      opacity: 0.15;
    }

    .icon-wrapper {
      position: absolute;
    }

    .icon {
      width: rem($width/1.5);
      height: rem($width/1.5);
    }
  }

  .ratio {
    font-size: rem(45);
    font-weight: bold;
    text-align: center;
    color: $black;
    margin: 0;
    margin-top: rem(12);
    margin-bottom: rem(12);
  }

  .title {
    font-size: rem(16);
    color: $grey;
    margin: 0;
  }
}
