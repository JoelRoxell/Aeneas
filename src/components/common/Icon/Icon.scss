.icon {
  position: relative;
  min-width: rem(16);
  min-height: rem(16);
  color: white;

  .badge {
    top: rem(-12);
    right: rem(-8);
    position: absolute;
    background: $tertiary;
    width: rem(20);
    height: rem(20);
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
    font-size: rem(10);
  }
}
