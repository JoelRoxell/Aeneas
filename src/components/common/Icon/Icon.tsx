import * as React from 'react'

import UserIcon from 'assets/svg/fill.svg'
import EyeWithLine from 'assets/svg/eye-with-line.svg'
import LogoSvg from 'assets/svg/logo.svg'
import LoaderSvg from 'assets/svg/loader.svg'

export const ICONS = {
  user: UserIcon,
  eye: EyeWithLine,
  logo: LogoSvg,
  loader: LoaderSvg
}

interface IconProps {
  name?: string
  width?: number
  height?: number
  fill?: string
  stroke?: string
  className?: string
  icon?: any
}

function Icon(props: IconProps) {
  let Component

  if (props.name) Component = ICONS[props.name]
  else Component = props.icon

  if (!Component) throw '<Icon />: Either name or icon must be set for an svg to render properly.'

  return <Component {...props} />
}

export default Icon
