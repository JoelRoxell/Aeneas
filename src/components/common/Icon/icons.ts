import EyeWithLine from 'assets/svg/eye-with-line.svg'
import LogoSvg from 'assets/svg/logo.svg'
import LoaderSvg from 'assets/svg/loader.svg'
import SystemSvg from 'assets/svg/system.svg'
import SettingsSvg from 'assets/svg/settings.svg'
import SensorSvg from 'assets/svg/sensor.svg'
import LogoutSvg from 'assets/svg/logout.svg'
import BellSvg from 'assets/svg/bell.svg'
import LightSvg from 'assets/svg/light.svg'
import HomeSvg from 'assets/svg/home.svg'
import DensitySvg from 'assets/svg/density.svg'
import UserSvg from 'assets/svg/user.svg'
import GrowZoneSvg from 'assets/svg/grow-zone.svg'

const ICONS = {
  eye: EyeWithLine,
  logo: LogoSvg,
  loader: LoaderSvg,
  system: SystemSvg,
  settings: SettingsSvg,
  sensor: SensorSvg,
  logout: LogoutSvg,
  bell: BellSvg,
  light: LightSvg,
  home: HomeSvg,
  density: DensitySvg,
  user: UserSvg,
  growZone: GrowZoneSvg
}

export default ICONS
