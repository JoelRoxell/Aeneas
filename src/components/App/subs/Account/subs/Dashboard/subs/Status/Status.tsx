import * as React from 'react'
import classNames from 'classnames'

import * as style from './Status.scss'
import Icon from 'components/common/Icon'
import posed from 'react-pose'
import Card from 'components/common/Card'
import { useDidMount } from 'hooks/useDidMount'

interface StatusProps {
  icon: React.ReactNode
  numerator: number
  denominator?: number
  className?: string
  title: string
  color: string
}

const StatusAnimation = posed.div({
  mounted: {
    opacity: 1
  },
  unMounted: {
    opacity: 0
  },
  initialPose: 'unMounted'
})

function Status(props: StatusProps) {
  const mounted = useDidMount()

  return (
    <StatusAnimation pose={mounted ? 'mounted' : 'unMounted'} className={props.className}>
      <Card className={classNames(style.status, props.className)}>
        <div className={style.circleWrapper}>
          <div className={style.circle} style={{ backgroundColor: props.color }} />

          <Icon
            icon={props.icon}
            className={style.iconWrapper}
            svgStyle={{ className: style.icon, fill: props.color }}
          />
        </div>

        {props.denominator ? (
          <p className={style.ratio}>{`${props.numerator} / ${props.denominator}`}</p>
        ) : (
          <p className={style.ratio}>{`${props.numerator}`}</p>
        )}

        <p className={style.title}>{props.title}</p>
      </Card>
    </StatusAnimation>
  )
}

export default Status
