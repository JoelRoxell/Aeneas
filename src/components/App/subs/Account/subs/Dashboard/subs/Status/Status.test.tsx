import * as React from 'react'
import { shallow } from 'enzyme'

import Status from './Status'
import { ICONS } from 'components/common/Icon'

describe('Status', () => {
  test('render', () => {
    const wrapper = shallow(
      <Status color="red" icon={ICONS.bell} numerator={3} denominator={4} title="A test" />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
