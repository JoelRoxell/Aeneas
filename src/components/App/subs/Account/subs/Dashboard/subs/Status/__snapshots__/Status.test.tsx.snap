// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status render 1`] = `
<ForwardRef
  pose="unMounted"
>
  <Card
    className="status"
  >
    <div
      className="circleWrapper"
    >
      <div
        className="circle"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      />
      <Icon
        className="iconWrapper"
        icon="default"
        svgStyle={
          Object {
            "className": "icon",
            "fill": "red",
          }
        }
      />
    </div>
    <p
      className="ratio"
    >
      3 / 4
    </p>
    <p
      className="title"
    >
      A test
    </p>
  </Card>
</ForwardRef>
`;
